pyat/at/physics/nonlinear.py

```diff
@@ -37,30 +53,46 @@ def tunes_vs_amp(ring: Lattice, amp: Optional[Sequence[float]] = None,
     """
 
     def _gen_part(ring, amp, dim, orbit, ld, nturns):
-        invx = numpy.array([[1/numpy.sqrt(ld['beta'][0]), 0],
-                            [ld['alpha'][0]/numpy.sqrt(ld['beta'][0]),
-                            numpy.sqrt(ld['beta'][0])]])
-
-        invy = numpy.array([[1/numpy.sqrt(ld['beta'][1]), 0],
-                            [ld['alpha'][1]/numpy.sqrt(ld['beta'][1]),
-                            numpy.sqrt(ld['beta'][1])]])
-        part = numpy.array([orbit, ] * len(amp)).T + 1.0e-6
+        invx = numpy.array(
```

Comment:
I would turn of black for formulas, this makes them very difficult to read

Reply:
non linear is restored from master w/o black.

Reply:
the new functions are moved to `magnet_tools.py`